cmd/sql/flow.go

```diff
@@ -0,0 +1,295 @@
+		return "", err
+	}
+
+	return projectDir, nil
+}
+
+func buildCommonVars() map[string]string {
+	vars := make(map[string]string)
+	if environment != "" {
+		vars["env"] = environment
+	}
+	if connection != "" {
+		vars["connection"] = connection
+	}
+	return vars
+}
+
+func getBaseMountDirs(projectDir string) ([]string, error) {
+	mountDir, err := createProjectDir(projectDir)
+	if err != nil {
+		return nil, err
+	}
+	mountDirs := []string{mountDir}
+	return mountDirs, nil
+}
+
+func executeAbout(cmd *cobra.Command, args []string) error {
+	vars := buildCommonVars()
+	mountDirs, err := getBaseMountDirs(projectDir)
+	if err != nil {
+		return err
+	}
+
+	cmdString := []string{cmd.Parent().Name(), cmd.Name()}
+	sql.CommonDockerUtil(cmdString, args, vars, mountDirs)
+
+	return nil
+}
+
+func executeVersion(cmd *cobra.Command, args []string) error {
+	vars := buildCommonVars()
+	mountDirs, err := getBaseMountDirs(projectDir)
+	if err != nil {
+		return err
+	}
+
+	cmdString := []string{cmd.Parent().Name(), cmd.Name()}
+	sql.CommonDockerUtil(cmdString, args, vars, mountDirs)
+
+	return nil
+}
+
+func executeInit(cmd *cobra.Command, args []string) error {
+	projectDir = "."
+	if len(args) > 0 {
+		projectDir = args[0]
+	}
+
+	vars := buildCommonVars()
+
+	mountDirs, err := getBaseMountDirs(projectDir)
+	args = mountDirs
+
+	if err != nil {
+		return err
+	}
+
+	if airflowHome != "" {
+		mountDirs = append(mountDirs, airflowHome)
+		vars["airflow-home"] = airflowHome
+	}
+
+	if airflowDagsFolder != "" {
+		mountDirs = append(mountDirs, airflowDagsFolder)
+		vars["airflow-dags-folder"] = airflowDagsFolder
+	}
+
+	cmdString := []string{cmd.Parent().Name(), cmd.Name()}
+	sql.CommonDockerUtil(cmdString, args, vars, mountDirs)
+
+	return nil
+}
+
+func executeValidate(cmd *cobra.Command, args []string) error {
+	if len(args) > 0 {
+		projectDir = args[0]
+	}
+
+	vars := buildCommonVars()
+
+	mountDirs, err := getBaseMountDirs(projectDir)
+	if err != nil {
+		return err
+	}
+	args = mountDirs
+
+	cmdString := []string{cmd.Parent().Name(), cmd.Name()}
+	sql.CommonDockerUtil(cmdString, args, vars, mountDirs)
+
+	return nil
+}
+
+func executeGenerate(cmd *cobra.Command, args []string) error {
+	if len(args) < 1 {
+		return sql.ArgNotSetError("workflow_name")
+	}
+
+	vars := buildCommonVars()
+	mountDirs, err := getBaseMountDirs(projectDir)
+	if err != nil {
+		return err
+	}
+
+	projectDir, err = getAbsolutePath(projectDir)
+	if err != nil {
+		return err
+	}
+	vars["project-dir"] = projectDir
+
+	cmdString := []string{cmd.Parent().Name(), cmd.Name()}
+	sql.CommonDockerUtil(cmdString, args, vars, mountDirs)
+
+	return nil
+}
+
+func executeRun(cmd *cobra.Command, args []string) error {
+	if len(args) < 1 {
+		return sql.ArgNotSetError("workflow_name")
+	}
+
+	vars := buildCommonVars()
+	mountDirs, err := getBaseMountDirs(projectDir)
+	if err != nil {
+		return err
+	}
+
+	projectDir, err = getAbsolutePath(projectDir)
+	if err != nil {
+		return err
```

Comment:
These lines of code are used across multiple functions in this file...Any chance we move it to a common function and re-use and not duplicate code